source/server/active_tcp_listener.cc

```diff
@@ -96,118 +75,6 @@ void ActiveTcpListener::updateListenerConfig(Network::ListenerConfig& config) {
   config_ = &config;
 }
 
-void ActiveTcpSocket::onTimeout() {
```

Comment:
This PR is very large and difficult to review.  Should you try to do a pure refactor of this class in a separate PR before extending its functionality?

Reply:
Sorry! This part doesn't bring logic change and the existing test cases cover each line of the function.

I am open to create a dedicated PR but I also want some opinions from @htuch and @mattklein123 to see if that PR is greatly reduce the load of review.

Reply:
I see references to internal connection type in some of the modified files, so at least some parts of this PR is not a pure refactor.

Reply:
Yeah, I think it would make review much more easier and faster if you can split this into two PRs:
1. The pure refactor, which hopefully we can process without too much bike shed and eliminate a large amount of the churn.
2. The new internal connection aspects. This is where the meat of the PR is, and where we should be able to spend the bulk of the review time.

Reply:
@antoniovicente @htuch Is #16947 a smooth refactor to you?

Reply:
I'm looking through it right now.  It is not a trivial refactor.  I see several class renames and new uses of the curiously recurring template pattern for not immediately clear reasons.